span.go

```diff
@@ -47,6 +35,46 @@ type Span interface {
 	// Suitable for serializing over the wire, etc.
 	TraceContext() TraceContext
 
+	// Log() records `data` to this Span at `timestamp`.
+	//
+	// See LogData for semantic details.
+	Log(timestamp time.Time, data LogData)
+
+	// Info() is equivalent to
+	//
+	//   Log(time.Now(), LogData{Message: fmt.Sprint(args...)})
+	//
+	Info(args ...interface{})
+
+	// Infof() is equivalent to
+	//
+	//   Log(time.Now(), LogData{Message: fmt.Sprintf(format, args...)})
```

Comment:
this is a bit confusing. Seems like these logging methods do not allow structured payloads, like a json, if it's always going to be transformed into a combined message after executing sprintf. Or does this allow lazy evals?